### DISCUSSION TITLE
Error when compiling LIS

### DISCUSSION THREAD

**Discussion by evanr1232** [NASA-LIS/LISF#1288] [Q&A] [answered]
Hello,

I'm getting the following error when trying to compile LIS:

ifort: command line warning #10434: option '-nomixed_str_len_arg' use with underscore is deprecated; use '-nomixed-str-len-arg' instead
mpiifort -c  -O2 -u -traceback -fpe0 -nomixed_str_len_arg -names lowercase -convert big_endian -assume byterecl -DSPMD -DHIDE_SHR_MSG -DNO_SHR_VMATH -DIFC -DLINUX -I/discover/nobackup/projects/lis/libs/sles-12.3/esmf/8.1.1_intel-2021.4.0_impi-2021.4.0/mod/modO/Linux.intel.64.intelmpi.default -DUSE_INCLUDE_MPI -I/discover/nobackup/projects/lis/libs/sles-12.3/ecCodes/2.23.0_intel-2021.4.0/include/ -I/discover/nobackup/projects/lis/libs/sles-12.3/netcdf/4.8.1_intel-2021.4.0/include/ -I/discover/nobackup/projects/lis/libs/sles-12.3/hdfeos2/2.20v1.00_intel-2021.4.0/include/ -I/discover/nobackup/projects/lis/libs/sles-12.3/hdf4/4.2.15_intel-2021.4.0/include/ -I/discover/nobackup/projects/lis/libs/sles-12.3/hdf5/1.12.1_intel-2021.4.0/include/ -DLIS_JULES -I./ -I../core/ -I../params/gfrac/SPORTDaily/ -I../params/gfrac/VIIRSDaily/ -I../surfacemodels/land/vic.4.1.1/ -I../surfacemodels/land/vic.4.1.1/physics/ -I../surfacemodels/land/vic.4.1.2.l/ -I../surfacemodels/land/vic.4.1.2.l/physics/ -I../surfacemodels/land/awral.6.0.0/physics/ -I../surfacemodels/land/rdhm.3.5.6/ -I../surfacemodels/land/rdhm.3.5.6/sac/ -I../surfacemodels/land/rdhm.3.5.6/frz/ -I../surfacemodels/land/rdhm.3.5.6/snow17/ -I../surfacemodels/land/rdhm.3.5.6/sachtet/ ../core/LIS_coreMod.F90

I noticed that [this discussion](https://github.com/NASA-LIS/LISF/discussions/1052) had the same issue, and I tried changing the environmental variables $FC and $CC to mpiifort and mpicc, but this did not work. It was also suggested to add the path to my Intel MPI installation's bin directory to $PATH, and I believe it already does: 

/discover/nobackup/projects/lis/libs/sles-12.3/netcdf/4.8.1_intel-2021.4.0/bin:/discover/nobackup/projects/lis/libs/sles-12.3/ecCodes/2.23.0_intel-2021.4.0/bin:/usr/local/other/git/2.24.0/libexec/git-core:/usr/local/other/git/2.24.0/bin:/usr/local/toolworks/2019/toolworks/memoryscape.2019.0.4/bin:/usr/local/toolworks/2019/toolworks/totalview.2019.0.4/bin:/usr/local/intel/oneapi/2021/mpi/2021.4.0/libfabric/bin:/usr/local/intel/oneapi/2021/mpi/2021.4.0/bin:/usr/local/intel/oneapi/2021/mkl/2021.4.0/bin/intel64:/usr/local/intel/oneapi/2021/debugger/10.2.4/gdb/intel64/bin:/usr/local/intel/oneapi/2021/compiler/2021.4.0/linux/bin/intel64:/usr/local/intel/oneapi/2021/compiler/2021.4.0/linux/bin:/usr/local/intel/oneapi/2021/compiler/2021.4.0/linux/ioc/bin:/usr/local/other/gcc/11.2.0/bin:/usr/local/bin:/usr/bin:/bin:/usr/bin/X11:/usr/games:/opt/ibutils/bin:/usr/lib/mit/bin:/usr/lib/mit/sbin:/opt/gnome/bin:/usr/slurm/bin

Please let me know if there's any known way to fix this error! Thank you!

**Comment by dmocko** [comment, marked as answer]
Hi @evanr1232 

I'm not sure I see an error in your post.  I only see the warning, and it should still compile even with the warning.  If you want to get rid of this warning, please do the following.

After you run the "configure" script, edit your LISF/lis/configure.lis file.  Replace the two instances of "-nomixed_str_len_arg" with "-nomixed-str-len-arg".  Then compile.

If there's another error you are seeing, please let us know.
